src/detect-quic-version.c

```diff
@@ -0,0 +1,261 @@
+/* Copyright (C) 2021 Open Information Security Foundation
+ *
+ * You can copy, redistribute or modify this Program under the terms of
+ * the GNU General Public License version 2 as published by the Free
+ * Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * version 2 along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+ * 02110-1301, USA.
+ */
+
+/**
+ *
+ * Implements the quic.version
+ */
+
+#include "suricata-common.h"
+#include "conf.h"
+#include "detect.h"
+#include "detect-parse.h"
+#include "detect-engine.h"
+#include "detect-engine-content-inspection.h"
+#include "detect-engine-uint.h"
+#include "detect-quic-version.h"
+#include "util-byte.h"
+#include "util-unittest.h"
+#include "rust.h"
+
+#ifdef UNITTESTS
+static void DetectQuicVersionRegisterTests(void);
+#endif
+
+static int quic_version_id = 0;
+
+static int DetectQuicVersionMatch(DetectEngineThreadCtx *det_ctx, Flow *f, uint8_t flags,
+        void *state, void *txv, const Signature *s, const SigMatchCtx *ctx);
+static int DetectQuicVersionSetup(DetectEngineCtx *, Signature *, const char *);
+void DetectQuicVersionFree(DetectEngineCtx *de_ctx, void *);
+
+static int DetectEngineInspectQuicVersionGeneric(DetectEngineCtx *de_ctx,
+        DetectEngineThreadCtx *det_ctx, const struct DetectEngineAppInspectionEngine_ *engine,
+        const Signature *s, Flow *f, uint8_t flags, void *alstate, void *txv, uint64_t tx_id);
+
+/**
+ * \brief Registration function for quic.version: keyword
+ */
+void DetectQuicVersionRegister(void)
+{
+    sigmatch_table[DETECT_AL_QUIC_VERSION].name = "quic.version";
+    sigmatch_table[DETECT_AL_QUIC_VERSION].desc = "match Quic version";
+    sigmatch_table[DETECT_AL_QUIC_VERSION].url = "/rules/quic-keywords.html#quic-version";
+    sigmatch_table[DETECT_AL_QUIC_VERSION].AppLayerTxMatch = DetectQuicVersionMatch;
+    sigmatch_table[DETECT_AL_QUIC_VERSION].Setup = DetectQuicVersionSetup;
+    sigmatch_table[DETECT_AL_QUIC_VERSION].Free = DetectQuicVersionFree;
+#ifdef UNITTESTS
+    sigmatch_table[DETECT_AL_QUIC_VERSION].RegisterTests = DetectQuicVersionRegisterTests;
+#endif
+
+    DetectAppLayerInspectEngineRegister2("quic.version", ALPROTO_QUIC, SIG_FLAG_TOSERVER, 1,
+            DetectEngineInspectQuicVersionGeneric, NULL);
+
+    quic_version_id = DetectBufferTypeGetByName("quic.version");
```

Comment:
@victorjulien do we need this ?
This is not a sticky buffer, so why would we need `DetectBufferTypeGetByName ` ?

Reply:
"Buffer" is a bit of a misnomer. Its registering a inspection list id. You can see that the id is used in `DetectQuicVersionSetup` below

Reply:
Thanks Victor, what is an inspection list id ?